physx.js

```diff
@@ -1273,13 +1273,53 @@ const physxWorker = (() => {
     );
     allocator.freeAll();
   };
+
+  w.marchingCubes = (dims, potential, shift, scale) => {
+    let allocator = new Allocator();
+
+    const dimsTypedArray = allocator.alloc(Int32Array, 3);
+    dimsTypedArray.set(dims);
+
+    const potentialTypedArray = allocator.alloc(Float32Array, potential.length);
+    potentialTypedArray.set(potential);
+
+    const shiftTypedArray = allocator.alloc(Float32Array, 3);
+    shiftTypedArray.set(shift);
+
+    const scaleTypedArray = allocator.alloc(Float32Array, 3);
+    scaleTypedArray.set(scale);
+
+    const outputBuffer = moduleInstance._doMarchingCubes(
+      dimsTypedArray.byteOffset,
+      potentialTypedArray.byteOffset,
+      shiftTypedArray.byteOffset,
+      scaleTypedArray.byteOffset
+    );
+
+    allocator.freeAll();
+
+    const head = outputBuffer / 4;
+
+    const positionCount = moduleInstance.HEAP32.subarray(head, head + 1)[0];

```

Comment:
`.subarray` aliases. If the user does not copy this, it will be corrupted next call. Should we copy?

Reply:
If `subarray()` is aliases, I think we need to copy to `Array`, not `TypedArray`

Reply:
Why not copy to `TypedArray`?

Reply:
You could literally just call `.slice` instead.